pkg/controllers/build/container_controller.go

```diff
@@ -129,6 +154,70 @@ func (r *ContainerReconciler) resolveTargetImage(ctx context.Context, log logr.L
 	return image, nil
 }
 
+func (r *ContainerReconciler) resolveDigestReference(ctx context.Context, log logr.Logger, ref name.Reference, container *buildv1alpha1.Container) (string, error) {
+	keychain, err := r.constructKeychain(ctx, log, container)
+	if err != nil {
+		return "", err
+	}
+
+	auth, err := keychain.Resolve(ref.Context().Registry)
+	if err != nil {
+		log.Error(err, "unable to resolve auth for registry", "registry", ref.Context().RegistryStr())
+		return "", err
+	}
+
+	img, err := remote.Image(ref, remote.WithAuth(auth))
+	if err != nil {
+		log.Error(err, "failed to read image", "image", ref.String())
+		return "", err
+	}
+
+	digest, err := img.Digest()
+	if err != nil {
+		log.Error(err, "failed to get image digest", "image", ref.String())
+		return "", err
+	}
+
+	return fmt.Sprintf("%s@%s", ref.Context().Name(), digest), nil
+}
+
+func (r *ContainerReconciler) constructKeychain(ctx context.Context, log logr.Logger, container *buildv1alpha1.Container) (gauthn.Keychain, error) {
+	var serviceAccount v1.ServiceAccount
+	if err := r.Get(ctx, types.NamespacedName{Namespace: container.Namespace, Name: riffBuildServiceAccount}, &serviceAccount); err != nil {
```

Comment:
We need to define RBAC policies and an Informer to be watch ServiceAccounts and Secrets. This currently happens to work because other controllers are already consumes these resources, but we should as well in case this reconciler is ever run in isolation, or the other reconciler is removed.

For the RBAC policy, we need to add at line ~50:
```
// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=get;list;watch
```

For the Informer, we need to add in SetupWithManager, between the `For` and `Complete`:
```
		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.Funcs{}).
		Watches(&source.Kind{Type: &corev1.ServiceAccount{}}, handler.Funcs{}).
```